x-pack/plugins/apm/public/context/chart_pointer_event/use_chart_pointer_event_context.tsx

```diff
@@ -16,14 +20,36 @@ export function useChartPointerEventContext() {
     throw new Error('Missing ChartPointerEventContext provider');
   }
 
-  const { pointerEvent } = context;
+  const { pointerEventTargetRef } = context;
   const chartRef = React.createRef<Chart>();
+  const requestIdRef = useRef(0);
+  const updatePointerEventHandler = useCallback(
+    (event: Event) => {
+      cancelAnimationFrame(requestIdRef.current);
```

Comment:
no need to make multiple updates between screen repaints, this cancels any pending updates so only the latest event will be dispatched to the chart component.